GTO: updating docs (#235)
iterative/mlem.ai: content/docs/gto/user-guide/index.md

GTO helps you build an Artifact Registry out of your Git repository. It creates
annotated [Git tags](https://git-scm.com/book/en/v2/Git-Basics-Tagging) with a
[special format](#git-tags-format) in their message, and manages an
`artifacts.yaml` file. Since committing large files to Git is not a good
practice, you should consider not committing your artifacts to Git. Instead,
[use DVC](/doc/gto/user-guide/dvc), Git-lfs, or any method to commit pointers to
the binary files instead.

> ## Annotations in artifacts.yaml

Reviewer: p.s. see also https://github.com/iterative/mlem.ai/pull/235 wrt the following paragraph. Feel free to apply the edits here instead.

[in a CI/CD system](#acting-in-ci-cd) downstream. But for more advanced cases,
we should codify them in the registry itself.

To keep this metadata, GTO uses `artifacts.yaml` file. Commands like
`gto annotate` and `gto remove` are used to modify it, while `gto describe`
helps get them when they're needed.

<admon type="tip">

An example of `artifacts.yaml` can be found in
[example-gto](https://github.com/iterative/example-gto/blob/main/artifacts.yaml)
repo.

</admon>

## Getting artifacts downstream

We may need to get a specific artifact version (latest or currently in stage) in
production environment to use it. GTO itself doesn't provide a way to download
an artifact, but you can use [DVC](/doc/user-guide/dvc) or
[MLEM](/doc/user-guide/mlem) to do that.

In all cases, you'll need to figure out the Git revision that you need (note
that in CI it's already defined for you - e.g. it's `GITHUB_REF` env var in
Github Actions):

```cli

> # getting the Git reference for the latest version
> $ gto show churn@greatest --ref
> churn@v3.1.1

Reviewer: ```suggestion
$ gto show churn@greatest --ref # greatest = latest ver.
churn@v3.1.1
```

Author: Let me fix it in https://github.com/iterative/gto/issues/306 and use `@latest` instead

> # for a model version in stage, that will be
> $ gto show churn#prod --ref
> churn@v3.0.0

Reviewer: ```suggestion
$ gto show churn#prod --ref # by assigned stage
churn@v3.0.0
```

```

> Beside the right revision, you may need artifact `path`. If
> [annotated](#annotations-in-artifactsyaml), it can be discovered by
> `gto describe`:
> 
> ```cli
> $ gto describe churn --rev churn@v3.0.0 --path
> models/churn.pkl

Reviewer: ````suggestion
You may need the artifact's file path. If
[annotated](#annotations-in-artifactsyaml), it can be discovered with
`gto describe`:

```cli
$ gto describe churn --rev churn@v3.0.0 --path
models/churn.pkl
````

[in a CI/CD system](#acting-in-ci-cd) downstream. But for more advanced cases,
we should codify them in the registry itself.

> To keep this metadata, GTO uses `artifacts.yaml` file. Commands like
> `gto annotate` and `gto remove` are used to modify it, while `gto describe`
> helps get them when they're needed.

Reviewer: Copy edit
```suggestion
To keep this metadata, GTO uses a human-readable `artifacts.yaml` file. The
`gto describe`, `gto annotate`, and `gto remove` commands are used to display and
manage it's contents.
```

<admon type="tip">

> An example of `artifacts.yaml` can be found in
> [example-gto](https://github.com/iterative/example-gto/blob/main/artifacts.yaml)
> repo.

Reviewer: Copy edit
```suggestion
An example `artifacts.yaml` can be found
[in the `example-gto` repo](https://github.com/iterative/example-gto).
```

$ GTO_EMOJIS=false gto show
```

> ## Git tags format

Author: No, it's Git tag name. E.g. to register a version, you need a Git tag called `model@v0.0.1`, annotated with some text. It could be annotated with any text, but without a text annotation it would be a `lightweight` tag, not an `annotated` one. For example, see https://github.com/iterative/example-gto/tags

<img width="342" alt="image" src="https://user-images.githubusercontent.com/6797716/204767470-3e7a0dc2-5827-426d-806b-333be38365b3.png">

`churn@v3.1.1` is a tag name and `Registering artifact churn version v3.1.1` is a Git tag message.

> ## Git tags format

Reviewer: > It could be annotated with any text, but without a text annotation it would be a lightweight tag... 

I see. May I ask why does GTO ignore lightweight tags?

> ## Git tags format

Author: Sure, we had a discussion about it here https://github.com/iterative/gto/issues/127
